client/src/components/Sharing/Sharing.vue

```diff
@@ -0,0 +1,469 @@
+                    this.assignItem(response.data);
+                    this.item.username_and_slug = `${this.itemSlugParts[0]}${newSlug}`;
+                })
+                .catch((error) => this.addError(error.response.data.err_msg));
+        },
+        onImportable(importable) {
+            if (importable) {
+                this.setSharing(this.actions.enable_link_access);
+            } else {
+                this.item.published = false;
+                this.setSharing(this.actions.disable_link_access);
+            }
+        },
+        onPublish(published) {
+            if (published) {
+                this.item.importable = true;
+                this.setSharing(this.actions.publish);
+            } else {
+                this.setSharing(this.actions.unpublish);
+            }
+        },
+        getModel() {
+            this.ready = false;
+            axios
+                .get(`${getAppRoot()}api/${this.pluralNameLower}/${this.id}/sharing`)
+                .then((response) => this.assignItem(response.data))
+                .catch((error) => this.addError(error.response.data.err_msg));
+        },
+        setUsername() {
+            const Galaxy = getGalaxyInstance();
+            axios
+                .put(`${getAppRoot()}api/users/${Galaxy.user.id}/information/inputs`, {
+                    username: this.newUsername || "",
+                })
+                .then((response) => {
+                    this.hasUsername = true;
+                    this.getModel();
+                })
+                .catch((error) => this.addError(error.response.data.err_msg));
+        },
+        setSharing(action, user_id, share_option) {
+            let user_ids = undefined;
+            if (user_id) {
+                if (user_id.includes(",")) {
+                    user_ids = user_id.replace(/ /g, "").split(",");
+                } else user_ids = [user_id];
+            }
+
+            if (
+                action === this.actions.share_with &&
+                user_ids &&
+                this.item.users_shared_with.some(({ email }) => user_ids.includes(email))
+            ) {
+                this.addError(
+                    `You already shared this ${this.model_class} with
+                    ${user_ids.filter((user) => this.item.users_shared_with.some(({ email }) => email === user))}`
+                );
+                return;
+            }
+
+            const data = {
+                user_ids: user_ids,
+                share_option: share_option ? share_option : undefined,
+            };
+            return axios
+                .put(`${getAppRoot()}api/${this.pluralNameLower}/${this.id}/${action}`, data)
+                .then(({ data }) => {
+                    this.errors = [];
+                    this.assignItem(data);
+                    if (data.extra && data.extra.can_share) this.shareWithEmail = "";
+                    else this.shareWithEmail = user_id || "";
```

Comment:
I think this might be more readable as either a ternary statement of a regular if / else with curly brackets.

Reply:
I could swear there was another comment about this but I can't find it right now.  Anyway, xref #11925 as an option for standardizing this.

Reply:
I will wait for https://github.com/galaxyproject/galaxy/pull/11925 on this. I hope you're fine with this